src/Akka.Hosting/AkkaHostedService.cs

```diff
@@ -11,36 +12,42 @@ namespace Akka.Hosting
     /// <summary>
     /// INTERNAL API
     /// </summary>
-    internal sealed class AkkaHostedService : IHostedService
+    /// <remarks>
+    /// Open for modification in cases where users need fine-grained control over <see cref="Actor.ActorSystem"/> startup and
```

Comment:
Warning to users: there will be tears if you don't study our source code when you try to extend this, and we will not help you with whatever weird DI stuff you're trying to do. Best of luck.